core/embed/extmod/modtrezorcrypto/modtrezorcrypto-aesgcm.h

```diff
@@ -0,0 +1,213 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "py/objstr.h"
+
+#include "aes/aesgcm.h"
+#include "memzero.h"
+
+/// package: trezorcrypto.__init__
+
+/// class aesgcm:
+///     """
+///     AES-GCM context.
+///     """
+typedef struct _mp_obj_AesGcm_t {
+  mp_obj_base_t base;
+  gcm_ctx ctx;
+  enum {
+    STATE_INIT,
+    STATE_ENCRYPTING,
+    STATE_DECRYPTING,
+    STATE_FINISHED,
+    STATE_FAILED,
+  } state;
+} mp_obj_AesGcm_t;
+
+/// def __init__(self, key: bytes, iv: bytes) -> None:
+///     """
+///     Initialize the AES-GCM context for encryption or decryption.
+///     """
+STATIC mp_obj_t mod_trezorcrypto_AesGcm_make_new(const mp_obj_type_t *type,
+                                                 size_t n_args, size_t n_kw,
+                                                 const mp_obj_t *args) {
+  mp_arg_check_num(n_args, n_kw, 2, 2, false);
+  mp_obj_AesGcm_t *o = m_new_obj_with_finaliser(mp_obj_AesGcm_t);
+  o->base.type = type;
+  o->state = STATE_INIT;
+  mp_buffer_info_t key = {0}, iv = {0};
+  mp_get_buffer_raise(args[0], &key, MP_BUFFER_READ);
+  mp_get_buffer_raise(args[1], &iv, MP_BUFFER_READ);
+  if (key.len != 16 && key.len != 24 && key.len != 32) {
+    m_del_obj(mp_obj_AesGcm_t, o);
```

Comment:
Is `m_del_obj()` really needed here? I'm asking because if it is, then it is missing in many other places in our codebase (for example [here](https://github.com/trezor/trezor-firmware/blob/c9f043bd966411cdb0597b0f221ddefc46a5fc9d/core/embed/extmod/modtrezorcrypto/modtrezorcrypto-aes.h#L74))

Reply:
Here are the others https://github.com/trezor/trezor-firmware/pull/3625/commits/c0f3ad49fe78a65e167ee5b161eb7f558862aad2